src/TopApp.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+} from 'react-router-dom';
+import axios from 'axios';
+import { Container } from 'semantic-ui-react';
+import { ToastContainer } from 'react-toastify';
+import App from './App';
+// import PrivateRoute from './PrivateRoute';
+import { UserContext, userContext } from './userContext';
+import 'react-toastify/dist/ReactToastify.css';
+
+axios.defaults.baseURL = 'http://localhost:4000/api';
+
+export default function TopApp() {
+  // Example of setting isAuthenticated in userContext, This function is for demonstration.
+  function handleOnAuthenticated(username) {
+    userContext.isAuthenticated = true;
+    userContext.username = username;
+  }
+
+  return (
+    <Container fluid>
+      <Router>
```

Comment:
You can move Router to index.js and wrap whole app.

Reply:
I think this way it is cleaner. The router functionalities are in top app

Reply:
I'm not sure that naming `TopApp` is pretty good. Lets move main app to `App.js` component.